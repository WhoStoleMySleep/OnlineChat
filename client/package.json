{
  "name": "client",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "@apollo/client": "^3.6.9",
    "@apollo/react-hoc": "^4.0.0",
    "@faker-js/faker": "^7.3.0",
    "@reduxjs/toolkit": "^1.8.3",
    "@testing-library/jest-dom": "^5.16.4",
    "@testing-library/react": "^13.3.0",
    "@testing-library/user-event": "^14.2.1",
    "@types/jest": "^27.5.2",
    "@types/node": "^17.0.45",
    "@types/react": "^18.0.15",
    "@types/react-dom": "^18.0.6",
    "classnames": "^2.3.1",
    "date-fns": "^2.29.1",
    "mathjs": "^10.6.4",
    "push.js": "^1.0.12",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-redux": "^8.0.2",
    "react-scripts": "5.0.1",
    "sass": "^1.53.0",
    "typescript": "^4.7.4",
    "use-sound": "^4.0.1",
    "web-vitals": "^2.1.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test --coverage",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "@typescript-eslint/eslint-plugin": "^5.30.6",
    "@typescript-eslint/parser": "^5.30.6",
    "babel-plugin-istanbul": "^6.1.1",
    "eslint": "^8.20.0",
    "eslint-config-airbnb": "^19.0.4",
    "eslint-plugin-import": "^2.26.0",
    "eslint-plugin-jest": "^26.6.0",
    "eslint-plugin-jsx-a11y": "^6.6.0",
    "eslint-plugin-react": "^7.30.1",
    "eslint-plugin-react-hooks": "^4.6.0",
    "react-test-renderer": "^18.2.0"
  }
}
