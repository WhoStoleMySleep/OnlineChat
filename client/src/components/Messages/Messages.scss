.messages {
  max-width: 1000px;
  width: 100%;
  margin: 0 auto;
  background: rgba(255, 255, 255, 0.05);
  box-shadow: 1px 1px 12px rgba(0, 0, 0, 0.1);
  border-radius: 4px;
  padding-top: 10px;
  box-sizing: border-box;
  transition-duration: 150ms;

  &:focus-within {
    border-radius: 18px;
  }
}

@media all and (min-width: 1440px) {
  .messages {
    max-width: none;
    width: 70%;
  }
}

@media all and (max-width: 1020px) {
  .messages {
    border-radius: 0;
    box-shadow: none;
    background: none;
    max-height: calc(100vh - 10px - 48px);
  }
}
