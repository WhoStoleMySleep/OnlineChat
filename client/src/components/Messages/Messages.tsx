import React from 'react';
import MessagesInput from '../MessagesInput/MessagesInput';
import MessagesList from '../MessagesList/MessagesList';
import styles from './Messages.module.scss';

function Messages() {
  return (
    <div className={styles.messages}>
      <MessagesList />
      <MessagesInput />
    </div>
  );
}

export default Messages;
