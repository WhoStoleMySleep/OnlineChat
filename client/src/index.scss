body {
  margin: 0;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
    'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  background: linear-gradient(
    120deg,
    rgba(23, 190, 187, 1),
    rgba(240, 166, 202, 1)
  );
  background-repeat: no-repeat;
  background-size: 100% 100%;
  height: 100vh;
  overflow-x: hidden;
}
