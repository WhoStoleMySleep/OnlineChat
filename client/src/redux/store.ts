import { configureStore } from '@reduxjs/toolkit';
import author from './componentReducers/login';
import messages from './componentReducers/messages';

export default configureStore({
  reducer: {
    messages,
    author
  }
}); 