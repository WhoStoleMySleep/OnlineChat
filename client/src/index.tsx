import {
  ApolloClient,
  ApolloProvider,
  HttpLink,
  InMemoryCache,
  split,
} from '@apollo/client';
import React from 'react';
import { createRoot } from 'react-dom/client';
import { Provider } from 'react-redux';
import App from './App';
import './index.scss';
import store from './redux/store';
import { WebSocketLink } from '@apollo/client/link/ws';
import { getMainDefinition } from '@apollo/client/utilities';

const httpLink = new HttpLink({
  uri: 'https://adasdasdasdasdasdsdfsdfsdfsdfs.herokuapp.com/graphql',
});

const wsLink = new WebSocketLink({
  uri: 'wss://adasdasdasdasdasdsdfsdfsdfsdfs.herokuapp.com/graphql',
  options: {
    reconnect: true,
  },
});

const splitLink = split(
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === 'OperationDefinition' &&
      definition.operation === 'subscription'
    );
  },
  wsLink,
  httpLink
);

const client = new ApolloClient({
  link: splitLink,
  cache: new InMemoryCache(),
});

const container = document.getElementById('root')!;
const root = createRoot(container);

root.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <Provider store={store}>
        <App />
      </Provider>
    </ApolloProvider>
  </React.StrictMode>
);
