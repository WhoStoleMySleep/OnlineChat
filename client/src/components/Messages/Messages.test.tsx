import '@testing-library/jest-dom';
import {
  render, screen
} from '@testing-library/react';
import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { Provider } from 'react-redux';
import Messages from './Messages';
import { MESSAGES_QUERY } from '../../GraphQl.queries';
import store from '../../redux/store';

const date = new Date().toString();

const mocks = [{
  request: {
    query: MESSAGES_QUERY,
  },
  result: {
    data: {
      id: '123', text: '123', author: '321', date
    }
  }
}];

describe('Message', () => {
  it('Message snapshot', async () => {
    render(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Provider store={store}>
          <Messages />
        </Provider>
      </MockedProvider>
    );

    const messages = screen.getByTestId('Messages');

    expect(messages).toMatchSnapshot();
  });
});
