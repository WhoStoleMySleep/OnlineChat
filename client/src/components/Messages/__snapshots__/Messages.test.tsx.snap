// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Message snapshot 1`] = `
<div
  class="messages"
  data-testid="Messages"
>
  <ul
    class="messages-list"
  />
  <form
    action=""
    class="messages-input"
    data-testid="form"
  >
    <input
      class="messages-input__input"
      data-testid="input"
      disabled=""
      placeholder="Enter your message"
      value=""
    />
    <input
      data-testid="submitInput"
      hidden=""
      type="submit"
    />
  </form>
</div>
`;
